src/overlays/actors/ovl_Obj_Mine/z_obj_mine.c

```diff
@@ -24,131 +53,1180 @@ ActorInit Obj_Mine_InitVars = {
+
+            this->waypointIndex = 0;
+            this->waypointCount = path->count - 1;
+            this->waypoints = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+
+            ObjMine_Path_MoveToWaypoint(this, this->waypointIndex);
+            ObjMine_Path_SetupMove(this);
+        }
+        Matrix_SetTranslateRotateYXZ(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
+                                     &this->actor.shape.rot);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+        Collider_UpdateSpheres(0, &this->collider);
+        this->actor.floorHeight = BgCheck_EntityRaycastFloor5(&play->colCtx, &this->actor.floorPoly, &dummyBgId,
+                                                              &this->actor, &this->actor.world.pos);
+    } else {
+        s32 linkCount = OBJMINE_GET_LINK_COUNT(&this->actor);
+
+        this->actor.update = ObjMine_AirWater_Update;
+        this->actor.uncullZoneScale = 150.0f + (linkCount * (LINK_SIZE * 1.75f));
+        this->actor.uncullZoneDownward = 150.0f + (linkCount * (LINK_SIZE * 1.75f));
+        ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 45.0f);
+        this->actor.shape.shadowAlpha = 140;
+
+        if (type == OBJMINE_TYPE_AIR) {
+            this->actor.draw = ObjMine_Air_Draw;
+            ObjMine_Air_InitChain(this, linkCount);
+            this->actor.world.pos.y = -ATTACH_OFFSET - (linkCount * LINK_SIZE) + this->actor.home.pos.y;
+            ObjMine_Air_InitCollider(this, linkCount);
+            func_800B4AEC(play, &this->actor, 0.0f);
+            if (linkCount == 0) {
+                ObjMine_Air_SetupStationary(this);
+            } else {
+                ObjMine_Air_SetupChained(this);
+            }
+        } else {
+            this->actor.draw = ObjMine_Water_Draw;
+            ObjMine_Water_InitChain(this, linkCount);
+            this->actor.world.pos.y = ATTACH_OFFSET + (linkCount * LINK_SIZE) + this->actor.home.pos.y;
+            ObjMine_Water_InitCollider(this, linkCount);
+            if (linkCount == 0) {
+                ObjMine_Water_SetupStationary(this);
+            } else {
+                ObjMine_Water_SetupChained(this);
+            }
+        }
+    }
+}
+
+void ObjMine_Destroy(Actor* thisx, PlayState* play) {
+    ObjMine* this = THIS;
+    // PlayState* play = play2; // must be in this order to match with recast
+
+    Collider_DestroyJntSph(play, &this->collider);
+}
+
+void ObjMine_Path_SetupStationary(ObjMine* this) {
+    this->actionFunc = ObjMine_Path_Stationary;
+}
+
+void ObjMine_Path_Stationary(ObjMine* this, PlayState* play) {
+}
+
+void ObjMine_Path_SetupMove(ObjMine* this) {
+    this->actor.flags |= ACTOR_FLAG_10;
+    this->actionFunc = ObjMine_Path_Move;
+}
+
+void ObjMine_Path_Move(ObjMine* this, PlayState* play) {
+    Actor* thisx = &this->actor; // pos and velocity need this, though it can replace all `this->actor`s and match
```

Comment:
I would say just use `thisx` for all of them than.

Reply:
I'd prefer to use `this` as much as possible, but if others prefer using `thisx` wherever possible when it has to be used at all I'll change it.

Reply:
That's usually the way we do it in this sort of situation, I think?